packages/lisk-p2p/src/peer_pool.ts

```diff
@@ -196,7 +207,22 @@ export class PeerPool extends EventEmitter {
 	}
 
 	private _handleGetAllPeersRequest(request: P2PRequest): void {
-		request.end(this.getAllPeerInfos());
+		const peerSelectionParams: PeerOptions = {
+			lastBlockHeight: this._nodeInfo ? this._nodeInfo.height : 0,
+		};
+
+		request.end({
+			success: true,
+			peers: this.selectPeers(peerSelectionParams).map((peer: Peer) => {
```

Comment:
maybe it's better to move this calculation above and assign to a const variable. Will be easier to debug I guess